File: test_yahoo.js
const fetch = require('node-fetch');

async function test(symbol) {
    const url = `https://query1.finance.yahoo.com/v8/finance/chart/${symbol}?interval=1d&range=1d`;
    try {
        const res = await fetch(url);
        const data = await res.json();
        const price = data.chart?.result?.[0]?.meta?.regularMarketPrice;
        console.log(symbol, '->', price);
    } catch (e) {
        console.log(symbol, '-> ERROR', e.message);
    }
}

(async () => {
    await test('RELIANCE.NS');
    await test('GOLD.MC'); // let's see if this works
    await test('MGC=F'); // Micro Gold
    await test('CL=F');
    await test('^DJI');
})();

File: test_yahoo.py
import urllib.request
import json

def test(symbol):
    url = f"https://query1.finance.yahoo.com/v8/finance/chart/{symbol}?interval=1d&range=1d"
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode())
            price = data.get('chart', {}).get('result', [{}])[0].get('meta', {}).get('regularMarketPrice')
            print(f"{symbol} -> {price}")
    except Exception as e:
        print(f"{symbol} -> ERROR {e}")

test('RELIANCE.NS')
test('GOLD.MC')
test('MGC=F')
test('CL=F')
test('^DJI')
test('^GSPC')
test('^NDX')
